### PR TITLE
Make error reports into a plugin

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary
* Turns key elements of error reporting into a plugin, to allow reporting to be deactivated
* Adds reporting of task errors to distributed error reporting
* Does some minor refactor to standardize naming between the core app and the plugin, rename the model to ErrorReport and recreate the migrations

## References
Fixes [#12473](https://github.com/learningequality/kolibri/issues/12473)

## Reviewer guidance

…

----

## Testing checklist

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/core/error_reports/middleware.py [hunk 1]
```diff
@@ -72,13 +76,13 @@ def process_exception(self, request, exception):
             "request_info": get_request_info(request),
             "server": get_server_info(request),
             "packages": get_packages(),
-            "python_version": get_python_version(),
+            "python_version": ".".join(str(v) for v in version_info[:3]),
```
Comment (thesujai) [inline kolibri/core/error_reports/middleware.py]: Maybe you missed this one, should call the method instead as it is also mocked in test_middleware

Comment (rtibbles) [reply kolibri/core/error_reports/middleware.py]: Hrm, weird - I patched that, but I guess it didn't take!

Comment (rtibbles) [reply kolibri/core/error_reports/middleware.py]: Updated!

File: kolibri/core/error_reports/test/test_models.py [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+from django.test import TestCase
+from django.utils import timezone
+
+from ..constants import BACKEND
+from ..constants import FRONTEND
+from ..constants import TASK
+from kolibri.core.error_reports.models import ErrorReport
+
+
+class ErrorReportsTestCase(TestCase):
```
Comment (akolson) [inline kolibri/core/error_reports/test/test_models.py]: No blocker, but for clarity's sake,  we should probably rename this to `ErrorReportTestCase` as the tests are specific to the `ErrorReport` model and not the plugin `error_reports`.

Comment (rtibbles) [reply kolibri/core/error_reports/test/test_models.py]: Will update!

Comment (rtibbles) [reply kolibri/core/error_reports/test/test_models.py]: Updated!

**Review by akolson — APPROVED** [review]
Code changes look correct to me! Thanks @rtibbles

**Comment by akolson** [comment]
@rtibbles is the docs failure a known issue?

**Comment by rtibbles** [comment]
Yes - it's fixed on develop.

**Pull request merged**